# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1266409880 -3600
#      Wed Feb 17 13:31:20 2010 +0100
# Node ID e0767a0965f1bb550d7479ddf3dd30ab73880008
# Parent  83fa590b8a09d9318b3c649901e156717d9a6c2a
fix some stat functions with integers

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-02-17  Jaroslav Hajek  <highegg@gmail.com>
+
+	* statistics/base/center.m: Convert integer inputs to doubles.
+
 2010-02-17  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/interp2: Slightly optimize linear interpolation.
 
 2010-02-12  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/interp2: Remove debug stub.
 
diff --git a/scripts/statistics/base/center.m b/scripts/statistics/base/center.m
--- a/scripts/statistics/base/center.m
+++ b/scripts/statistics/base/center.m
@@ -36,14 +36,18 @@ function retval = center (x, dim)
     print_usage ();
   endif
 
   if (nargin < 2)
     dim = [find(size (x) != 1, 1), 1](1); # First non-singleton dim.
   endif
   n = size (x, dim);
 
+  if (isinteger (x))
+    x = double (x);
+  endif
+
   if (n == 0)
     retval = x;
   else
     retval = bsxfun (@minus, x, sum (x, dim) / n);
   endif
 endfunction
